### PR TITLE
[New Rule] CyberArkPas promotion rules

### LINKED ISSUE #1304: [New Rule] Rules for the CyberArk Privileged Access Security Integration
## Description
Create rules leveraging telemetry from the [cyberarkpas](https://github.com/elastic/integrations/tree/master/packages/cyberarkpas) integration.

### Target indexes
* `filebeat-*`
* `logs-cyberarkpas.audit*`

### Additional requirements
integration or filebeats module

### Tested ECS Version
1.10.0

### References
* [elastic docs](https://www.elastic.co/guide/en/beats/filebeat/master/exported-fields-cyberarkpas.html)
* [integration](https://github.com/elastic/integrations/tree/master/packages/cyberarkpas)
* [beats module](https://github.com/elastic/beats/tree/master/x-pack/filebeat/module/cyberarkpas)
* CyberArk PAS [audit events](https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/Latest/en/Content/PASREF/Vault%20Audit%20Action%20Codes.htm?TocPath=Administration%7CReferences%7C_____3) and [docs](https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/Latest/en/Content/HomeTilesLPs/LP-Tile1.htm?TocPath=Get%20Started%7C_____0)

## Example Data
Lots of data [here](https://github.com/elastic/beats/tree/master/x-pack/filebeat/module/cyberarkpas/audit/test)

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
resolves #1304 

## Summary
CyberArkPas is a recently introduced [integration](https://github.com/elastic/integrations/tree/master/packages/cyberarkpas) in Kibana. Per the [vendor](https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/10.10/en/Content/PASIMP/Introducing-the-Privileged-Account-Security-Solution-Intro.htm):

>The CyberArk's Privileged Access Security (PAS) solution is a full life-cycle solution for managing the most privileged accounts and SSH Keys in the enterprise. It enables organizations to secure, provision, manage, control and monitor all activities associated with all types of privileged identities

This introduces two promotion rules for the `"logs-cyberarkpas.audit*"` events:
* `CyberArk Privileged Access Security Error` - this will alert on error level events, which are alertable events usually associated with failed auth or other security risks
* `CyberArk Privileged Access Security Recommended Monitor` - this will alert on any of a list of event codes which the vendor recommends monitoring

These are the first _true_ promotion rules and so tuning recommendations were made under `false_positives` and additional information added to the investigation guide

`event.dataset` was not added until 1.1.0 for [7.14](https://github.com/elastic/integrations/blob/master/packages/cyberarkpas/manifest.yml#L11) and so should not be backported

## Details

<details>

To test, sample data was ingested from the integration [tests](https://github.com/elastic/integrations/tree/master/packages/cyberarkpas/data_stream/audit/_dev/test/pipeline)

#### CyberArk Privileged Access Security Error

All included `event.type:error` events
image

#### CyberArk Privileged Access Security Recommended Monitor

Screenshot of some of the triggered rules from the sample data
image

Some of these may be very noisy (SQL command) based on monitoring configurations, so a note was added on how to tune under `false_positives`

</details>

### CODE REVIEW

**Review by threat-punter — APPROVED** [review]

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+The cyberarkpas Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.
+
+## Triage and analysis
+
+This is a promotion rule for CyberArk error events, which are alertable events per the vendor.
+Consult vendor documentation on interpreting specific events.
+"""
+references = [
+    "https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/Latest/en/Content/PASREF/Vault%20Audit%20Action%20Codes.htm?tocpath=Administration%7CReferences%7C_____3",
```
Comment (rw-access) [inline rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:26]: do we need the `?tocpath=Administration%7CReferences%7C_____3` param?

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+description = "Identifies the occurrence of a CyberArk Privileged Access Security (PAS) error level audit event."
+false_positives = ["To tune this rule, add exceptions to exclude any event.code which should not trigger this rule."]
+from = "now-30m"
+index = ["filebeat-*", "logs-cyberarkpas.audit*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "CyberArk Privileged Access Security Error"
+note = """## Config
+
+The cyberarkpas Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.
```
Comment (rw-access) [inline rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:18]: these names (`cyberarkpas`) aren't made easily visible to the user, I would recommend spelling out the full name.

also, this wording feels a little more natural to me
```suggestion
The cyberarkpas Fleet integration, Filebeat module, or similarly structured data is required for proper function of this rule.
```

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+[metadata]
+creation_date = "2021/06/23"
+maturity = "production"
+updated_date = "2021/06/23"
+min_stack_version = '7.14.0'
+
+[rule]
+author = ["Elastic"]
+description = "Identifies the occurrence of a CyberArk Privileged Access Security (PAS) error level audit event."
+false_positives = ["To tune this rule, add exceptions to exclude any event.code which should not trigger this rule."]
```
Comment (rw-access) [inline rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:10]: Please explain either here or in the description what `event.code` means. By excluding an `event.code` does that mean that I'm suppressing a rule on that side?

**Review by rw-access — COMMENTED** [review]
Let's start a new directory structure and move these to `integrations/cyberark`.

That'll make it easier when we get around to https://github.com/elastic/detection-rules/issues/1033

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+[metadata]
+creation_date = "2021/06/23"
+maturity = "production"
+updated_date = "2021/06/23"
+min_stack_version = '7.14.0'
```
Comment (rw-access) [inline rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:5]: this is exciting

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml (hunk 1 above)
Comment (brokensound77) [reply rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:26]: I know it is awful, but yes it is a specific reference

**Comment by brokensound77** [comment]
> Let's start a new directory structure and move these to `integrations/cyberark`.
> 
> That'll make it easier when we get around to #1033

I like this idea, but where do we draw the line? Most of these also exist as filebeat modules. We have folders for AWS, Azure, etc. - so should we move all those too? Maybe if an integration exists, we move it there (which would _most_ help prep for #1033)

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml (hunk 2 above)
Comment (brokensound77) [reply rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:18]: to me that is _less_ accurate and feels like it implies the logic or operation of the rule changes based on data.

I am not against changing it, but the most important thing we want to imply here is to have the right data, however it gets in

Comment (brokensound77) [reply rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:18]: ++ to spelling out

**Comment by rw-access** [comment]
Yeah, I'm thinking that AWS, Azure, Okta, etc. would all have the same structure. It's fine with me if you want to punt on that

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+[rule]
+author = ["Elastic"]
+description = "Identifies the occurrence of a CyberArk Privileged Access Security (PAS) error level audit event."
```
Comment (brokensound77) [inline rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:9]:
```suggestion
description = """Identifies the occurrence of a CyberArk Privileged Access Security (PAS) error level audit event. The 
event.code correlates to the CyberArk Vault Audit Action Code."""
```

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml (hunk 2 above)
Comment (brokensound77) [reply rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:18]:
```suggestion
The CyberArk Privileged Access Security (PAS) Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.
```

File: rules/cyberark/privilege_escalation_cyberarkpas_recommended_events_to_monitor_promotion.toml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+"""
+false_positives = ["To tune this rule, add exceptions to exclude any event.code which should not trigger this rule."]
+from = "now-30m"
+index = ["filebeat-*", "logs-cyberarkpas.audit*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "CyberArk Privileged Access Security Recommended Monitor"
+note = """## Config
+
+The cyberarkpas Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.
```
Comment (brokensound77) [inline rules/cyberark/privilege_escalation_cyberarkpas_recommended_events_to_monitor_promotion.toml:21]:
```suggestion
The CyberArk Privileged Access Security (PAS) Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.```

File: rules/cyberark/privilege_escalation_cyberarkpas_recommended_events_to_monitor_promotion.toml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+creation_date = "2021/06/23"
+maturity = "production"
+updated_date = "2021/06/23"
+min_stack_version = '7.14.0'
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the occurrence of a CyberArk Privileged Access Security (PAS) non-error level audit event which is
+recommended for monitoring by the vendor.
```
Comment (brokensound77) [inline rules/cyberark/privilege_escalation_cyberarkpas_recommended_events_to_monitor_promotion.toml:11]:
```suggestion
recommended for monitoring by the vendor. The event.code correlates to the CyberArk Vault Audit Action Code.
```

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml (hunk 3 above)
Comment (brokensound77) [reply rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:12]: add to description b3ddf345ef10c058615be1585b5994e0103ca40b

**Comment by brokensound77** [comment]
> brokensound77 added backport: auto 

Will actually need to backport now that the 7.14 branch is made

**Comment by rw-access** [comment]
> Will actually need to backport now that the 7.14 branch was created

Since #1332 was made, `backport: auto` "just works" so it would filter it out.

**Comment by brokensound77** [comment]
> Yeah, I'm thinking that AWS, Azure, Okta, etc. would all have the same structure. It's fine with me if you want to punt on that

#1360

File: rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml (hunk 4 above)
Comment (rw-access) [reply rules/cyberark/privilege_escalation_cyberarkpas_error_audit_event_promotion.toml:5]: getting ahead of the lint change
```suggestion
min_stack_version = "7.14.0"
```

**Review by rw-access — APPROVED** [review]
small nit, otherwise LGTM

**Pull request merged**